app/services/purchase_order_creator.rb

```diff
@@ -0,0 +1,46 @@
+class PurchaseOrderCreator < ApplicationService
+  def initialize(purchase_params)
+    @address_params = purchase_params[:address] || {}
```

Comment:
```suggestion
    @address_params = purchase_params.fetch(:address, {})
```

Reply:
Essa modificação não funcionou. Ela fez quebrar todos os testes oiginalmente criados pelo Wender.

` 1) Purchases POST /create when gateway is PayPal and cart exists and there is a user present in cart does not create a guest user
     Failure/Error: address_1: @address_params[:address_1],
     
     NoMethodError:
       undefined method `[]' for nil:NilClass`

Reply:
Isso pode ser efeito colateral de outro erro. Vou dar uma olhadinha. Mas aparentemente o @address_params está `nil`!

Reply:
Resolvemos em pairing.